### PR TITLE
[API v3] Maintenance Mode

### PR DESCRIPTION
**Pull request by SabreCat**
Fixes https://github.com/HabitRPG/habitrpg/issues/7167
### Changes

Implements a "maintenance mode" on the v3 server, which renders a maintenance page for HTML requests and returns an error object for other requests.

Still needed before launch:
- [ ] ~~Get maintenance end date/time from environment variable and display it in user's local time~~
- [x] Correct URL for information page about the maintenance
- [x] Additional link to Habitica Twitter for updates
- [ ] ~~If possible, display a readonly view of users' tasks -- this will require letting some requests through the middleware~~

---

UUID:

### CODE REVIEW

File: website/server/middlewares/api-v3/maintenanceMode.js [hunk 1]
```diff
@@ -0,0 +1,24 @@
...
+module.exports = function maintenanceMode (req, res, next) {
+  if (MAINTENANCE_MODE !== 'true') return next();
+
+  const MAINTENANCE_END = new Date('Sat May 21 2016 12:45 UTC');
+
+  if (req.headers && req.headers.accept && req.headers.accept.indexOf('text/html') !== -1) {
+    return res.status(503).render('../../../views/static/maintenance', {
+      env: res.locals.habitrpg,
+      t: res.t,
+      maintenanceEndDate: MAINTENANCE_END.toLocaleDateString(),
```
Comment (paglias) [inline website/server/middlewares/api-v3/maintenanceMode.js]: here the locale hour will be that of the server, you should pass maintenanceStartDate date object to the client and call toLocaleDateString() there

Comment (paglias) [inline website/server/middlewares/api-v3/maintenanceMode.js]: in client side javascript

File: website/server/middlewares/api-v3/maintenanceMode.js [hunk 2]
```diff
@@ -0,0 +1,24 @@
...
+
+module.exports = function maintenanceMode (req, res, next) {
+  if (MAINTENANCE_MODE !== 'true') return next();
+
+  const MAINTENANCE_END = new Date('Sat May 21 2016 12:45 UTC');
+
+  if (req.headers && req.headers.accept && req.headers.accept.indexOf('text/html') !== -1) {
+    return res.status(503).render('../../../views/static/maintenance', {
+      env: res.locals.habitrpg,
+      t: res.t,
```
Comment (paglias) [inline website/server/middlewares/api-v3/maintenanceMode.js]: for res.t to work you have to first run the getUserLanguage() middleware

File: website/server/middlewares/api-v3/maintenanceMode.js [hunk 3]
```diff
@@ -0,0 +1,24 @@
...
+      maintenanceEndDate: MAINTENANCE_END.toLocaleDateString(),
+      maintenanceEndTime: MAINTENANCE_END.toLocaleTimeString(),
```
Comment (paglias) [inline website/server/middlewares/api-v3/maintenanceMode.js]: passing maintenanceEndTime twice

Comment (SabreCat) [inline website/server/middlewares/api-v3/maintenanceMode.js]: One's the date string and the other's the time, but I can do that on the client anyway per your other comment!

File: website/server/middlewares/api-v3/maintenanceMode.js (hunk 2 above)
Comment (SabreCat) [inline website/server/middlewares/api-v3/maintenanceMode.js]: `app.use(attachTranslateFunction)` comes before `app.use(maintenanceMode)`; is there something else I need?

File: website/server/middlewares/api-v3/maintenanceMode.js (hunk 3 above)
Comment (paglias) [inline website/server/middlewares/api-v3/maintenanceMode.js]: sorry I didn't notice that!

**Comment by paglias** [comment]
pushed a commit so that in maintenance mode it doesn't connect to MongoDB

**Pull request merged**